extensions/common/sql/sql-lease/src/test/java/org/eclipse/edc/sql/lease/SqlLeaseContextTest.java

```diff
@@ -0,0 +1,121 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       SAP SE - tests for bugfix (pass correct lease id for deletion)
+ *
+ */
+
+package org.eclipse.edc.sql.lease;
+
+import org.eclipse.edc.spi.monitor.Monitor;
+import org.eclipse.edc.transaction.local.LocalTransactionContext;
+import org.eclipse.edc.transaction.spi.TransactionContext;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.time.Clock;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.when;
+
+class SqlLeaseContextTest {
```

Comment:
generally I  tend to agree that unit tests should be preferred, but since we already have the `PostgresLeaseContextTest`, please just move the test methods over there. 
As a neat side effect, will make the tests much clearer

Reply:
1. Are the integration test results also part of code coverage? Will it reduce the code coverage if I don't have a unit test?
2. Should I move the tests to `PostgresLeaseContextTest` (there are currently no tests here) or to `LeaseContextText` (abstract class that has all the tests)?

Reply:
Coverage it's just a number that's not that important in term of code quality, so I wouldn't bother about it, things needs to be tested properly.

`LeaseContextTest` is the right place to put such tests

Reply:
@ndr-brt I will no longer have access to the `connection` object as it is defined in the child class (`PostgresLeaseContextTest`). Is it okay to add an abstract method for the query/param verification in the `LeaseContextTest` and implement the actual check in the child class?

Reply:
Agreed. But the leases are orphaned and not associated with transfer process or negotiations anymore as the update statement will associate it with the latest lease. The `getLease` call will always return only 1. I will move the test. Thank you.

Reply:
What "update statement"? The error is in the "delete", right? If this mechanism won't permit the existence of two leases on the same entity (because it does not make sense) then it would be easier to test, without having to check in the table with a specific query.

Reply:
Tests moved to `PostgresLeaseContextTest` as agreed.